gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveDataset.java

```diff
@@ -0,0 +1,294 @@
+    } else {
+      return new Path(pattern, this.table.getTableName());
+    }
+  }
+
+  /**
+   * Parse a location string from a {@link StorageDescriptor} into an actual list of globs.
+   */
+  private List<Path> parseLocationIntoPaths(String sdLocation) {
+    List<Path> locations = Lists.newArrayList();
+
+    if (sdLocation == null) {
+      return locations;
+    }
+
+    for (String location : this.splitter.split(sdLocation)) {
+      locations.add(new Path(location));
+    }
+    return locations;
+  }
+
+  /**
+   * Finds all files read by the table and generates CopyableFiles. The semantics are as follows:
+   * 1. Find all valid {@link StorageDescriptor}. If the table is partitioned, the {@link StorageDescriptor} of the base
+   *    table will be ignored, and we will instead process the {@link StorageDescriptor} of each partition.
+   * 2. For each {@link StorageDescriptor} find all files referred by it.
+   * 3. Generate a {@link CopyableFile} for each file referred by a {@link StorageDescriptor}.
+   * 4. If the table is partitioned, create a file set for each partition.
+   *
+   * <p>
+   *   The target locations of data files for this table depend on the values of the resolved table root (e.g.
+   *   the value of {@link #COPY_TARGET_TABLE_ROOT} with tokens replaced) and {@link #RELOCATE_DATA_FILES_KEY}:
+   *   * if {@link #RELOCATE_DATA_FILES_KEY} is true, then origin file /path/to/file/myFile will be written to
+   *     /resolved/table/root/<partition>/myFile
+   *   * otherwise, if the resolved table root is defined (e.g. {@link #COPY_TARGET_TABLE_ROOT} is defined in the
+   *     properties), we define:
+   *     origin_table_root := the deepest non glob ancestor of table.getSc().getLocation() iff getLocation() points to
+   *                           a single glob. (e.g. /path/to/*&#47;files -> /path/to). If getLocation() contains none
+   *                           or multiple globs, job will fail.
+   *     relative_path := path of the file relative to origin_table_root. If the path of the file is not a descendant
+   *                      of origin_table_root, job will fail.
+   *     target_path := /resolved/table/root/relative/path
+   *     This mode is useful when moving a table with a complicated directory structure to a different base directory.
+   *   * if {@link #COPY_TARGET_TABLE_ROOT} is not defined in the properties, then the target is identical to the origin
+   *     path.
+   * </p>
+   */
+  @Override public Collection<CopyableFile> getCopyableFiles(FileSystem targetFs, CopyConfiguration configuration)
+      throws IOException {
+
+    List<CopyableFile> copyableFiles = Lists.newArrayList();
+
+    try(AutoReturnableObject<IMetaStoreClient> client = new AutoReturnableObject<>(this.clientPool)) {
+      if (isPartitioned()) {
+        List<Partition> partitions = client.get().listPartitions(this.table.getDbName(), this.table.getTableName(),
+            Short.MAX_VALUE);
```

Comment:
Hmm, yes, the -1 is documented for other similar methods, but their javadocs are terrible. I will change it.